pkg/pillar/cmd/zedrouter/server.go

```diff
@@ -542,6 +546,14 @@ func (hdl kubeConfigHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 		return
 	}
+
+	if binary.Size(kubeConfig) > KubeconfigFileSizeLimitInBytes {
```

Comment:
Actually it belongs to the previous modifications, but can we not use `ioutil.ReadAll` here, but some kind of limited reader? It looks like the way to OOM from user app.

Reply:
Thanks for pointing that out!
I added a 35KB limit to the reader.

Reply:
I thought the limit on the reader was there in the code which introduced the handler @zed-rishabh .